### ISSUE TITLE
Unit test failure in bitemporal store

### ISSUE THREAD

**Issue by bmoscon** [manahl/arctic#59]
def test_last_update(bitemporal_library):
        bitemporal_library.update('spam', ts1, as_of=dt(2015, 1, 1))
        bitemporal_library.update('spam', ts1, as_of=dt(2015, 1, 2))
>       assert bitemporal_library.read('spam').last_updated == dt(2015, 1, 2)
E       assert Timestamp('2015-01-02 05:00:00') == datetime.datetime(2015, 1, 2, 0, 0)
E        +  where Timestamp('2015-01-02 05:00:00') = BitemporalItem(symbol='spam', library=u'arctic_test.TEST', data=              ...:06:11.040   3.0, metadata=None, last_updated=Timestamp('2015-01-02 05:00:00')).last_updated
E        +    where BitemporalItem(symbol='spam', library=u'arctic_test.TEST', data=              ...:06:11.040   3.0, metadata=None, last_updated=Timestamp('2015-01-02 05:00:00')) = <bound method BitemporalStore.read of <arctic.store.bitemporal_store.BitemporalStore object at 0x7f1c2da62650>>('spam')
E        +      where <bound method BitemporalStore.read of <arctic.store.bitemporal_store.BitemporalStore object at 0x7f1c2da62650>> = <arctic.store.bitemporal_store.BitemporalStore object at 0x7f1c2da62650>.read
E        +  and   datetime.datetime(2015, 1, 2, 0, 0) = dt(2015, 1, 2)

tests/integration/store/test_bitemporal_store.py:60: AssertionError

Basically we're comparing Timestamp('2015-01-02 05:00:00') to Timestamp('2015-01-02 00:00:00')

Not sure why last updated is reporting 05:00:00 as the timestamp. Is this because I'm not running my machine on GMT or is this some other anomaly? Doesn't appear version related (I tried with 1.9.2/0.16.2 and latest, same behavior)

**Comment by AdrianTeng** [comment]
Hi Bryant, this does look like timezone issue.

The `as_of` column is timezone aware:

```
        local_tz = mktz()
        if not as_of:
            as_of = dt.now()
        if as_of.tzinfo is None:
            as_of = as_of.replace(tzinfo=local_tz)
```
If `mktz()` and `dt.now()` is reporting inconsistent data (say, `dt.now()` gives Boston time and `mktz()` gives London timezone). Would you mind confirming those two functions' output on your machine?

**Comment by bmoscon** [comment]
```>>> from arctic.date import mktz
>>> mktz()
tzfile('/usr/share/zoneinfo/America/New_York')
>>> import datetime
>>> datetime.datetime.now().tzinfo
>>> datetime.datetime.now()
datetime.datetime(2015, 12, 19, 8, 19, 45, 540301)
>>> datetime.datetime.now().tzinfo is None
True
>>> 
```

**Comment by AdrianTeng** [comment]
Right so we have two problems here.
- The RHS of the assert statement should have local tz info
- (!) Multi-datetime-index df loses timezone info after write->read

Will look into a fix

**Comment by bmoscon** [comment]
Is there a reason we don't just store everything at UTC and apply a timezone later?

**Comment by AdrianTeng** [comment]
It does! It's just the codepath not supporting multi-index fully (yet)
